crates/turbopack-ecmascript/src/references/external_module.rs

```diff
@@ -0,0 +1,226 @@
+use std::io::Write;
+
+use anyhow::{Context, Result};
+use turbo_tasks::Vc;
+use turbo_tasks_fs::{rope::RopeBuilder, FileContent, FileSystem, VirtualFileSystem};
+use turbopack_core::{
+    asset::{Asset, AssetContent},
+    chunk::{AsyncModuleInfo, ChunkItem, ChunkType, ChunkableModule, ChunkingContext},
+    ident::AssetIdent,
+    module::Module,
+    reference::ModuleReferences,
+    resolve::ExternalType,
+};
+
+use crate::{
+    chunk::{
+        EcmascriptChunkItem, EcmascriptChunkItemContent, EcmascriptChunkPlaceable,
+        EcmascriptChunkType, EcmascriptChunkingContext, EcmascriptExports,
+    },
+    references::async_module::{AsyncModule, OptionAsyncModule},
+    utils::StringifyJs,
+    EcmascriptModuleContent,
+};
+
+#[turbo_tasks::function]
+fn layer() -> Vc<String> {
+    Vc::cell("external".to_string())
+}
+
+#[turbo_tasks::value]
+pub struct ExternalModuleAsset {
+    pub request: String,
+    pub external_type: ExternalType,
+    pub import_externals: bool,
+}
+
+impl ExternalModuleAsset {
+    pub fn is_async(&self) -> bool {
+        self.external_type == ExternalType::EcmaScriptModule && self.import_externals
+    }
+}
+
+#[turbo_tasks::value_impl]
+impl ExternalModuleAsset {
+    #[turbo_tasks::function]
+    pub fn new(request: String, external_type: ExternalType, import_externals: bool) -> Vc<Self> {
+        Self::cell(ExternalModuleAsset {
+            request,
+            external_type,
+            import_externals,
+        })
+    }
+
+    #[turbo_tasks::function]
+    pub fn is_async_vc(&self) -> Vc<bool> {
+        Vc::cell(self.is_async())
+    }
+
+    #[turbo_tasks::function]
+    pub fn content(&self) -> Vc<EcmascriptModuleContent> {
```

Comment:
Instead of all the unwraps in this method, you could return a `Result` here:


```suggestion
    pub fn content(&self) -> Result<Vc<EcmascriptModuleContent>> {
```

Reply:
all the writes are infallible, but I guess there's no real difference